### PR TITLE
BAEL-1041 Introduction to Caffeine

### PR DESCRIPTION
**Pull request by retroq**

### CODE REVIEW

File: libraries/src/test/java/com/baeldung/caffeine/CaffeineUnitTest.java [hunk 1]
```diff
@@ -0,0 +1,156 @@
...
+import com.github.benmanes.caffeine.cache.AsyncLoadingCache;
+import com.github.benmanes.caffeine.cache.Cache;
+import com.github.benmanes.caffeine.cache.Caffeine;
+import com.github.benmanes.caffeine.cache.Expiry;
+import com.github.benmanes.caffeine.cache.LoadingCache;
+
+public class CaffeineUnitTest {
+
+    @Test
+    public void given_Cache_whenPopulate_thenValueStored() {
```
Comment (KevinGilmore) [inline libraries/src/test/java/com/baeldung/caffeine/CaffeineUnitTest.java]: Several of the test names begin with "given_".
Let's remove the trailing underscore. So this one, for example, would just be:
"givenCache_whenPopulate_thenValueStored"

Comment (retroq) [reply libraries/src/test/java/com/baeldung/caffeine/CaffeineUnitTest.java]: Done

File: libraries/src/test/java/com/baeldung/caffeine/CaffeineUnitTest.java [hunk 2]
```diff
@@ -0,0 +1,158 @@
...
+    @Test
+    public void givenCache_whenPopulate_thenValueStored() {
+        Cache<String, DataObject> cache = Caffeine.newBuilder()
+          .expireAfterWrite(1, TimeUnit.MINUTES)
+          .maximumSize(100)
+          .build();
+
+        String key = "A";
+        DataObject dataObject = cache.getIfPresent(key);
+        assertNull(dataObject);
```
Comment (KevinGilmore) [inline libraries/src/test/java/com/baeldung/caffeine/CaffeineUnitTest.java]: Always add a blank line between what is being tested and the assertion(s).

File: libraries/src/main/java/com/baeldung/caffeine/DataObject.java [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+    @Override
+    public String toString() {
+        return "DataObject{" +
+                "data='" + data + '\'' +
+                '}';
+    }
+
+    public static DataObject get(String data) {
+        objectCounter++;
+        System.out.println(String.format("Initializing DataObject#%d with data '%s'", objectCounter, data));
```
Comment (pivovarit) [inline libraries/src/main/java/com/baeldung/caffeine/DataObject.java]: That looks like an overkill, let's use some logger here that can interpolate strings

**Pull request merged**